src/tikzplotlib/_legend.py

```diff
@@ -78,7 +78,7 @@ def draw_legend(data, obj):
     if alignment:
         data["current axes"].axis_options.append(f"legend cell align={{{alignment}}}")
 
-    if obj._ncol != 1:
+    if obj._ncols != 1:
         data["current axes"].axis_options.append(f"legend columns={obj._ncol}")
```

Comment:
You'll have to change both:
```suggestion
    if obj._ncols != 1:
        data["current axes"].axis_options.append(f"legend columns={obj._ncols}")
```
otherwise if ncols > 1 this line would then fail.

This is a change in the internal details of matplotlib in version 3.6.0, which tikzplotlib unfortunately has to depend on. But by making this change, you break compatibility with all older versions. It would be more robust to change it as follows, to be compatible both with older versions and the new changes:
```suggestion
    try:
        ncols = obj._ncols
    except AttributeError:
        # backwards-compatibility with matplotlib < 3.6.0
        ncols = obj._ncol
    if ncols != 1:
        data["current axes"].axis_options.append(f"legend columns={ncols}")
```

Reply:
Thanks for the fix, this would also fit with matplotlib's policy on `_ncols` overriding `_ncol` whenever possible.